FrameworkInternals/transformDesign.py

```diff
@@ -107,6 +107,18 @@ def run_indent_tool(unindented_content, fout):
        unindented_content - stream of bytes (e.g. encoded string) to pass to the tool,
        fout - a file object to which the output will be written"""
     try:
+        astyle_version_check_process = subprocess.run(['astyle', '--version'],
+                                                          stderr=subprocess.PIPE,
+                                                          stdout=subprocess.PIPE,
+                                                          check=True)
+        # note to line below: whether version is printed in stdout or stderr unfortunately depends
+        # on astyle version ... ironic ;-)
+        astyle_version_text = astyle_version_check_process.stderr.decode('utf-8') + astyle_version_check_process.stdout.decode('utf-8')
+        if 'Artistic Style Version 3' not in astyle_version_text:
```

Comment:
Is this a built-in future problem ? I mean when astyle 4.0 comes out.

Maybe could regex the string for 'Artistic Style Version X', and barf if X <3 ?

Reply:
Hi Ben,

I have no idea how the version tag will look like with 4.X so I'd prefer not to make predictions.
In fact the primary reason this is added is because CC7 is defined as our baseline platform and astyle 2.X seems still kind of present there (might be my fault as it was me who 'populated' astyle). I think that once the baseline moves on from CC7 we could get rid of that check.